src/services/http-client.service.ts

```diff
@@ -42,44 +42,69 @@ export class HttpClientService {
         });
     }
 
-    public async updateData(url: string, profile: Profile, body: object): Promise<any> {
+    public async updateData(url: string, headers: Headers, body: object): Promise<any> {
         return new Promise<any>((resolve, reject) => {
-            request.put(url, this.makeOptions(profile, body), (err, res) => {
+            request.put(url, this.makeOptions(headers, body), (err, res) => {
                 this.handleResponse(res, resolve, reject);
             });
         });
     }
 
-    public async findAll(url: string, profile: Profile): Promise<any> {
+    public async findAll(url: string, headers: Headers): Promise<any> {
         return new Promise<any>((resolve, reject) => {
-            request.get(url, this.makeOptions(profile), (err, res) => {
+            request.get(url, this.makeOptions(headers, null), (err, res) => {
                 this.handleResponse(res, resolve, reject);
             });
         });
     }
 
-    private makeOptions(profile: Profile, body: object = {}): CoreOptions {
+    public async authenticate(profile: Profile): Promise<any> {
```

Comment:
add authenticate method to create headers.

Reply:
I really like this idea of making a simple request to check if the token should be used with the `Bearer` or `AppKey` prefix, however I think that making this decision before each command/http call is too much when using a profile. I would suggest adding a prefix property in the `Profile` interface and setting it to `Bearer` or `AppKey` when validating [here](https://github.com/celonis/content-cli/pull/71/files#diff-c8416a6cab692c74a7a6cd69256d4f2a0c9a96c7ac67628553eec415c5a68508R36), depending on the validation result. Then you could use the prefix from the profile in this class to build the Authorization header accordingly.

You also need to do a similar approach to this `authenticate` method to create the profile prefix [here](https://github.com/celonis/content-cli/blob/5ba40389fc62556a7530a9fefebf3ba71ab19ed6/src/services/profile.service.ts#L71), so command calls with env variables also work. The following command is an example of using env variables instead of profiles.
```
API_TOKEN=<some-token> TEAM_URL=<some-team-url> content-cli push package
```

Reply:
@promeris 
Thanks for your check. 
Regarding authentication by Profile option I got your point.

On the other hand regarding authentication by env variable, still I am confused.
If I understand your comment correctly, I should create new env variable AUTHORIZATION to set 'Bearer' or 'AppKey', then user should setup API_TOKEN / TEAM_URL / AUTHORIZATION before running content-cli. Do you agree ?

Reply:
The env variable authentication should be working the same as before with the TEAM_URL and API_TOKEN, where you can provide an Application or User token in the API_TOKEN variable. However you would need somehow set the new prefix property for the profile [here](https://github.com/celonis/content-cli/blob/master/src/services/profile.service.ts#L72), because we create a profile from the env variables. 

To achieve this you can do this approach of testing out the token by making a request to `/api/cloud` and check if the prefix to be used should be `Bearer` or `AppKey`. I hope this clears your confusion.

Reply:
@promeris 
I got your point. 
When env variables API_TOKEN / TEAM_URL are passed, then call `authenticate` method to build headers.
If env variables are not there, then use Profile file and get api_token, team_url and new attribute ('Bearer' or 'AppKey').

I will do update until next week.